### PR TITLE
Fixes for email notifications not sending share link in the body

### PR DESCRIPTION
**Pull request by TejasRGitHub**
### Feature or Bugfix
- Bugfix

### Detail
- Correcting place where the environment variable containing frontend URL is defined. Earlier it was in the AWSWorker lambda env but it is needed in the graphQL lambda
- Removing " which was causing issues when clicking on email address. Also lowercased words. 

### Relates
- https://github.com/data-dot-all/dataall/issues/1142

### Security
Please answer the questions below briefly where applicable, or write `N/A`. Based on
[OWASP 10](https://owasp.org/Top10/en/).

- Does this PR introduce or modify any input fields or queries - this includes
fetching data from storage outside the application (e.g. a database, an S3 bucket)? N/A
  - Is the input sanitized?
  - What precautions are you taking before deserializing the data you consume?
  - Is injection prevented by parametrizing queries?
  - Have you ensured no `eval` or similar functions are used?
- Does this PR introduce any functionality or component that requires authorization? N/A
  - How have you ensured it respects the existing AuthN/AuthZ mechanisms?
  - Are you logging failed auth attempts?
- Are you using or adding any cryptographic features? N/A
  - Do you use a standard proven implementations?
  - Are the used keys controlled by the customer? Where are they stored?
- Are you introducing any new policies/roles/users? N/A
  - Have you used the least-privilege principle? How?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: deploy/stacks/lambda_api.py [hunk 1]
```diff
@@ -91,14 +91,18 @@ def __init__(
         self.api_handler_dlq = self.set_dlq(f'{resource_prefix}-{envname}-graphql-dlq')
         api_handler_sg = self.create_lambda_sgs(envname, 'apihandler', resource_prefix, vpc)
         api_handler_env = {'envname': envname, 'LOG_LEVEL': 'INFO', 'REAUTH_TTL': str(reauth_ttl)}
+        # Check if custom domain exists and if it exists email notifications could be enabled. Create a env variable which stores the domain url. This is used for sending data.all share weblinks in the email notifications.
+        if custom_domain and custom_domain.get('hosted_zone_name', None):
+            api_handler_env['frontend_domain_url'] = f'https://{custom_domain.get("hosted_zone_name", None)}'
+        api_handler_role = self.create_function_role(envname, resource_prefix, 'graphql', pivot_role_name)
```
Comment (dlpzx) [inline deploy/stacks/lambda_api.py:97]: I think we do not need to create the `api_handler_role` outside of the custom_auth if-statement and we can keep the previous implementation `role=self.create_function_role(envname, resource_prefix, 'graphql', pivot_role_name),` in line 105

**Review by dlpzx — COMMENTED** [review]
Very minor change proposed. But it looks good, thanks @TejasRGitHub :)

File: deploy/stacks/lambda_api.py (hunk 1 above)
Comment (TejasRGitHub) [reply deploy/stacks/lambda_api.py]: Fixed!!

**Comment by TejasRGitHub** [comment]
Updated code.

**Review by dlpzx — APPROVED** [review]
Looks good!

**Pull request merged**